src/backend/distributed/utils/reference_table_utils.c

```diff
@@ -434,26 +434,13 @@ void
 DeleteAllReferenceTablePlacementsFromNodeGroup(int32 groupId)
 {
 	List *referenceTableList = CitusTableTypeIdList(REFERENCE_TABLE);
-	List *referenceShardIntervalList = NIL;
 
 	/* if there are no reference tables, we do not need to do anything */
 	if (list_length(referenceTableList) == 0)
 	{
 		return;
 	}
 
-	/*
-	 * We sort the reference table list to prevent deadlocks in concurrent
-	 * DeleteAllReferenceTablePlacementsFromNodeGroup calls.
-	 */
-	referenceTableList = SortList(referenceTableList, CompareOids);
-	if (ClusterHasKnownMetadataWorkers())
-	{
-		referenceShardIntervalList = GetSortedReferenceShardIntervals(referenceTableList);
-
-		BlockWritesToShardList(referenceShardIntervalList);
-	}
-
 	StringInfo deletePlacementCommand = makeStringInfo();
 	Oid referenceTableId = InvalidOid;
 	foreach_oid(referenceTableId, referenceTableList)
```

Comment:
but doesn't the metadata syncing below also require all nodes to be up?

Reply:
yes, there are going to be some follow-up PRs. So, if you prefer, we might defer merging these PRs until we have all of them ready. 

Mostly opening for ease of review each piece.